docs/source/frameworks/transformers.rst

```diff
@@ -87,6 +87,29 @@ such as tokenizers, preprocessors, and feature extractors, can also be saved as
 
 To load the pre-trained instances for testing and debugging, use :code:`bentoml.transformers.load_model` with the same tags.
 
+Starting from BentoML version 1.1.9, importing pre-trained Transformer models from Hugging Face has been further streamlined.
+You can choose to use the new ``bentoml.transformers.import_model`` function to import models directly into the BentoML Model Store without the overhead of loading them into memory.
+By contrast, ``bentoml.transformers.save_model`` necessitates prior model loading and can be resource-intensive for models with large weights.
+Here is an example of using the new function:
+
+.. code-block:: python
+    :caption: `download_model.py`
+
+    import bentoml
+    from transformers import AutoTokenizer
+
+    # Save the tokenizer with minimal memory overhead
+    tokenizer = AutoTokenizer.from_pretrained("t5-small")
+    bentoml.transformers.save_model('t5-small-tokenizer', tokenizer)
+
+    # Import the model without loading into memory, conserving memory
+    bentoml.transformers.import_model("t5-small-model", "t5-small")
+
+The ``bentoml.transformers.import_model`` function has two required parameters:
+
+* ``name``: The name of the model in the BentoML Model Store.
+* ``model_name_or_path``: This can be a string, a Hugging Face repository identifier (repo_id), or a directory path containing weights saved using ``transformers.AutoModel.save_pretrained`` (for example, ``./my_pretrained_directory/``).
+
```

Comment:
Hey Sherlock, everything looks good to me so far. I would like to mention that there is a small note that should be added:

At the moment, repositories that require the argument `trust_remote_code=True` to be set will actually fail to download. I am currently working on a change to fix this issue. 

The new changes should be merged this week as well. With the new changes, users will unfortunately need to load the model into memory by default for repositories that require `trust_remote_code=True`. To avoid loading into memory, users will have to set the `clone_repository=True` keyword argument.

Reply:
This change can probably summarized as follows:

NOTE: Importing from model repositories that use the keyword argument `trust_remote_code=True` will require loading the model into memory by default. To completely avoid loading the model into memory, we recommend using the keyword argument `clone_repository=True`. This downloads all the files in the repository instead of selectively picking certain model files by using the method `.from_pretrained` under the hood, avoiding any potential memory usage of the model weights.

Reply:
Thank you for the clarification. So `clone_repository=True` should be added like this?

```python
import bentoml

model = "sshleifer/distilbart-cnn-12-6"
task = "summarization"

# Import the model without loading into memory by cloning the entire repository
bentoml.transformers.import_model(
   name=task, 
   model_name_or_path=model,
   clone_repository=True,  # This will avoid loading the model into memory
   metadata=dict(model_name=model)
)

```

Reply:
`import_model` won't load the model into memory regardless `clone_repository` is set or not.

The model only load into memory if the models require to use `trust_remote_code=True` (for examples, any Yi AI models, Baichuan, ChatGLM models). In that case, setting `clone_repository=True` won't load such models into memory

Reply:
however `clone_repository=True` may increase the size of the downloaded model. Should we mention this?

Reply:
Both Aaron and Larme said are correct. For most cases, model will not load into memory. But some repositories require the use of `trust_remote_code=True` keyword and will load model to memory. To avoid this, we can set `clone_repository=True`, which will download all the repo files (hence the larger storage size requirement) but will not load model to memory. 

```
# Import trust_remote_code=True model without loading into memory by cloning the entire repository

bentoml.transformers.import_model(
   name=task, 
   model_name_or_path=model,
   trust_remote_code=True,
   clone_repository=True,  # This will avoid loading the model into memory
   metadata=dict(model_name=model)
)
```

Reply:
Thank you all. I just updated the docs based on your comments.

Reply:
> however `clone_repository=True` may increase the size of the downloaded model. Should we mention this?

Worth noting is we probably want to have an options for ppl to save safetensors vs legacy bin weights